diofant/polys/rootoftools.py

```diff
@@ -299,9 +303,52 @@ def _complexes_sorted(cls, complexes):
             return []
         cache = {}
 
+        def is_re_disjoint(u, v):
+            if u.conj != v.conj:
+                return True
+            else:
+                return u.bx <= v.ax or v.bx <= u.ax
+
+        res_cache = {}
+
+        def match_same_real_part(u, v):
+            assert u.domain == v.domain
+            dom = u.domain
+            if tuple(u.f) not in res_cache:
+                res_cache[tuple(u.f)] = dmp_resultant(dmp_permute(u.f1, [1, 0], 1, dom),
+                                                      dmp_permute(u.f2, [1, 0], 1, dom),
+                                                      1, dom)
+            if tuple(v.f) not in res_cache:
+                res_cache[tuple(v.f)] = dmp_resultant(dmp_permute(v.f1, [1, 0], 1, dom),
+                                                      dmp_permute(v.f2, [1, 0], 1, dom),
+                                                      1, dom)
+            g = dmp_gcd(res_cache[tuple(u.f)], res_cache[tuple(v.f)], 0, dom)
+            a, b = max(u.ax, v.ax), min(u.bx, v.bx)
```

Comment:
I think that `max` and `min` should be interchanged. There could be a single root in the intersection of the intervals but that could be the wrong real part if there are several roots in the union of the intervals.

Reply:
But comon roots should be in the intersection of isolation rectangles, isn't?  If there is any, of course.  So, this code refines initial bounds for real part, nothing more.  If there are some roots of ``g`` outside of this interval (namely, in the union, not in intersection) - following code expected to return ``False``.

Reply:
If it is known at this point that there is a single root in a rectangle, then I think this is ok.

Reply:
Yes, that should be known after [L327](https://github.com/skirpichev/diofant/blob/8cee0050df3574e22bce22b7769cea8e6747a4a5/diofant/polys/rootoftools.py#L327).  If we got something different (not a single root) - procedure will return ``False`` and refinement will be continued on [L351](https://github.com/skirpichev/diofant/blob/8cee0050df3574e22bce22b7769cea8e6747a4a5/diofant/polys/rootoftools.py#L351).